### DISCUSSION TITLE
Consider chainging the type of key in the Column interface to improve type safety.

### DISCUSSION THREAD

**Discussion by CriusNyx** [Comcast/react-data-grid#2466] [Ideas]
In your Column interface on line 47 of index.d.ts consider changing the type of key of the following.

`
key : keyof TRow
`

This will cause the compiler to throw a type error when developers use a key that is not supported by their TRow interface.

**Comment by nstepien** [comment]
Rows are left generic so the rows can be a nested structure, or an Immutable.js object for example.
